recordval.go

```diff
@@ -31,141 +29,96 @@ func (r *Value) Walk(rv Visitor) error {
 	return Walk(r.Type, r.Bytes, rv)
 }
 
-// Slice returns the encoded zcode.Bytes corresponding to the indicated
-// column or an error if a problem was encountered.
-func (r *Value) Slice(column int) (zcode.Bytes, error) {
+func (r *Value) nth(column int) zcode.Bytes {
 	var zv zcode.Bytes
 	for i, it := 0, r.Bytes.Iter(); i <= column; i++ {
 		if it.Done() {
-			return nil, ErrMissing
+			return nil
 		}
 		zv = it.Next()
 	}
-	return zv, nil
+	return zv
 }
 
 func (r *Value) Columns() []Column {
 	return TypeRecordOf(r.Type).Columns
 }
 
-// Value returns the indicated column as a Value.  If the column doesn't
-// exist or another error occurs, the nil Value is returned.
-func (r *Value) ValueByColumn(col int) Value {
-	zv, err := r.Slice(col)
-	if err != nil {
-		return Value{}
+func (v *Value) DerefByColumn(col int) *Value {
```

Comment:
We should call this `Index` and make it work for other types but later is fine for all that.